From e854060dbf65c17f92b6a5d4247bb1d7b77c90d1 Mon Sep 17 00:00:00 2001
From: Varsha Maira <vmaira@habana.ai>
Date: Fri, 14 Jun 2024 14:01:04 +0300
Subject: [PATCH] [FSW-418] fs1: bootrom: get hash for tpm pcr extend

This patch adds function to fetch the hash that
needs to be extended for a given TPM PCR index.

Change-Id: Ic5a6c5847c922d4ade0767d680c2235d0544c6f3
Signed-off-by: Varsha Maira <vmaira@habana.ai>
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index bf35916..f29a8f8 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -262,4 +262,43 @@
 void boot_rom_tpm_next_pcr_index_init(void)
 {
 	boot_rom_set_tpm_next_pcr_index(TPM2_SVN_PCR_IDX);
-}
\ No newline at end of file
+}
+
+static bool boot_rom_get_svn(__unused uint8_t *hash)
+{
+	/*
+	 * TODO add code to fetch SVN from littlefs
+	 * This code will be added once the capability to read
+	 * SVN from eFuse and store it in littlefs is integrated
+	 */
+	return false;
+}
+
+static bool boot_rom_get_manifest_hash(__unused uint8_t *hash)
+{
+	/*
+	 * TODO add code to fetch manifest hash from littlefs
+	 * This code will be added once the capability to compute
+	 * manifest hash and store it in littlefs is integrated
+	 */
+	return false;
+}
+
+bool boot_rom_get_hash_for_tpm_pcr(uint8_t tpm_pcr_index, uint8_t *hash)
+{
+	bool is_valid = false;
+
+	switch (tpm_pcr_index) {
+	case TPM2_SVN_PCR_IDX:
+		is_valid = boot_rom_get_svn(hash);
+		break;
+	case TPM2_MANIFEST_PCR_IDX:
+		is_valid = boot_rom_get_manifest_hash(hash);
+		break;
+	default:
+		is_valid = false;
+		break;
+	}
+
+	return is_valid;
+}
diff --git a/common/common.h b/common/common.h
index 97e9dcc..c75082b 100644
--- a/common/common.h
+++ b/common/common.h
@@ -28,5 +28,6 @@
 void boot_rom_set_tpm_next_pcr_index(uint8_t tpm_pcr_index);
 void boot_rom_tpm_next_pcr_index_init(void);
 uint32_t boot_rom_tpm_spi_baudr_get(void);
+bool boot_rom_get_hash_for_tpm_pcr(uint8_t tpm_pcr_index, uint8_t *hash);
 
 #endif /* _COMMON_H_ */
